src/System.Linq.Expressions/src/System/Linq/Expressions/Error.cs

```diff
@@ -1120,5 +1120,10 @@ internal static Exception NonAbstractConstructorRequired()
         {
             return new InvalidOperationException(Strings.NonAbstractConstructorRequired);
         }
+
+        internal static Exception InvalidProgram()
+        {
+            return new InvalidProgramException();
```

Comment:
InvalidProgramException is thrown when JIT chokes on some IL. I think, when this gets thrown from interpreter, most will assume we have hit a JIT bug. 

InvalidOperationException seems a better option.